### PR TITLE
feat(amazonq): Support more programming languages for workspace index

### PR DESCRIPTION
**Pull request by leigaol**
## Problem

C# and some other languages are not supported for workspace index.

## Solution

Add support for more programming languages.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazonq/.changes/next-release/Bug Fix-fe345557-1053-4096-a3f7-f1b11d982a49.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
```
Comment (hayemaxi) [inline packages/amazonq/.changes/next-release/Bug Fix-fe345557-1053-4096-a3f7-f1b11d982a49.json:2]: Sounds like this is a feature instead

File: packages/amazonq/.changes/next-release/Feature-fe345557-1053-4096-a3f7-f1b11d982a49.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Support more programming languages for workspace index"
```
Comment (justinmk3) [inline packages/amazonq/.changes/next-release/Feature-fe345557-1053-4096-a3f7-f1b11d982a49.json:3]: This is an opportunity to link to relevant docs pages or marketing landing pages (markdown is supported), if possible.

**Comment by justinmk3** [comment]
is this ready to merge? needs review from someone on Q team

**Review by zixlin7 — APPROVED** [review]

File: packages/amazonq/.changes/next-release/Feature-fe345557-1053-4096-a3f7-f1b11d982a49.json (hunk 2 above)
Comment (leigaol) [reply packages/amazonq/.changes/next-release/Feature-fe345557-1053-4096-a3f7-f1b11d982a49.json:3]: We do not have relevant doc pages for this.

**Pull request merged**